docs/source/schema-directives.md

```diff
@@ -1,179 +1,527 @@
 ---
 title: Schema directives
-description: Using custom Directive on FIELD_DEFINITION
+description: Implementing and using custom `@directive`s that transform schema types, fields, and arguments
 ---
 
-## Simple Directive example
+## Schema directives
 
-Let's take a look at how we can create `@upper` Directive to upper-case a string returned from resolve on Field
-[See a complete runnable example on Launchpad.](https://launchpad.graphql.com/p00rw37qx0)
+A _directive_ is an identifier preceded by a `@` character, optionally followed by a list of named arguments, which can appear after almost any form of syntax in the GraphQL query or schema languages.
 
-To start, let's grab the schema definition string from the `makeExecutableSchema` example [in the "Generating a schema" article](/tools/graphql-tools/generate-schema.html#example).
+The [GraphQL specification](http://facebook.github.io/graphql/October2016/#sec-Type-System.Directives) requires every server implementation to support at least two directives, `@skip(if: Boolean)` and `@include(if: Boolean)`, which can be used during query execution to conditionally omit or include certain fields. The [GraphQL.js reference implementation](https://github.com/graphql/graphql-js) provides one additional built-in [`@deprecated`](https://github.com/graphql/graphql-js/blob/master/src/type/directives.js) directive, which is useful for indicating that a field or `enum` value should no longer be used.
```

Comment:
I think talking about `@skip` and `@include` here is confusing since those appear on queries. People often confuse the two and they have very little to do with each other.

Often when people hear "schema directive" they think it's something that lets them implement a new directive they can write in their queries, which is not the case; so we should try not to confuse them further.

BTW a good place to link to is maybe the PR for adding the schema language to the spec: https://github.com/facebook/graphql/pull/90

Also, here's the deployed draft version that has it: http://facebook.github.io/graphql/draft/#sec-Type-System.Directives

It's pretty relevant:

<img width="679" alt="image" src="https://user-images.githubusercontent.com/448783/36748763-a6771e16-1bad-11e8-8b56-d74a9da961e6.png">